### DISCUSSION TITLE
macOS Sonoma npm start not working

### DISCUSSION THREAD

**Discussion by Kos-devUK** [community/community#68733] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Unfortunately I've updated my software to Sonoma.
Every time I enter the 'npm start' command on my React app directory, is loading but the app is not appearing on my browser. 
Any help will be appreciated 
<img width="1437" alt="Screenshot 2023-09-29 at 09 54 29" src="https://github.com/community/community/assets/129678217/e2a9dfdb-dfcf-42ed-9cdf-56b78f9a421c">

**Comment by krgraver** [comment]
I'm having the same issue, looking for help as well.

**Reply by Kos-devUK** [reply]
Are you having issues to start up your os environment as well ?
Every time Sonoma starts, next to the apple logo is running "Installer Progress".

**Reply by krgraver** [reply]
No issues on startup for me

**Reply by krgraver** [reply]
I was able to resolve the issue, you just have to update Xcode

**Reply by Kos-devUK** [reply]
well done, but I am not using Xcode
the whole update for me didn't work, next time I'll wait a month or two, with all the bug fixes

**Reply by peterekey** [reply]
Updating Xcode worked for me too. After I upgraded to Sonoma, I restarted my machine, and then I was prompted to update Xcode. After Xcode finished updating, `npm start` worked as expected.

**Comment by GianfrancoCaputo** [comment]
I was able to fix this by deleting the node_modules folder and running npm install again. Hope it helps

**Reply by trulstp** [reply]
this worked fro me as well

**Reply by vkaralewich** [reply]
Same. Makes sense, can't transfer node_modules computer to computer, and you're kind of doing that when upgrading to Sonoma.

**Reply by hassaanjamil** [reply]
Quick fix, worked for me, thanks mate.

**Reply by Rytis-J** [reply]
Had a problem with NPM / Vite not working properly. I installed the latest version of MacOS developer beta (MacOS Sequoia) and had the same issue and the solution worked with MacOS Sequoia as well.  Apparently Xcode older versions don't work with the latest MacOS and Xcode 16 beta is needed. After installing the new Xcode software and reinstalling node_modules I was able to fix the issue. 

Hats off to you sir!

**Comment by xtycoonfi** [comment]
Try this : 
- Check your node version => `node -v`
- If it is not the lateast => `nvm install --lts`
- Then delete node_modules => `rm -rf /node_modules`
- Reinstall packages => `npm i`
- Then => `npm run start`

Also, be sure you have `start` below `scripts`inside your packages.json file.

**Comment by Kos-devUK** [comment]
I found the issue (I think)...
I had the react app on my iCloud and the file wasn't downloading properly, (something went wrong with the update).
Then I deactivate the cloud and I run the app("npm start") from a back up folder and now it runs properly.
Thanks for the help though :)
